admin/public/src/actions/configs.jsx

```diff
@@ -18,32 +18,33 @@ export function loadConfig(config) {
 }
 
 export function fetchConfigs() {
-  return dispatch => dispatch({
-    [CALL_API]: {
-      endpoint: '/api/configs',
-      method: 'GET',
-      credentials: 'same-origin',
-      types: [CONFIGS_REQUEST, CONFIGS_FETCH_SUCCESS, CONFIGS_FAILURE]
+  return async dispatch => {
+    try {
+      dispatch({ type: CONFIGS_REQUEST });
+      const response = await fetch('/api/configs', { method: 'GET', credentials: 'same-origin' });
+      return dispatch({ type: CONFIGS_FETCH_SUCCESS, payload: await response.json() });
+    } catch (e) {
+      return dispatch({ type: CONFIGS_FAILURE });
```

Comment:
Should dispatch the error as well, so `{ type: CONFIGS_FAILURE, payload: e }`, otherwise it turns into an infinite loop

Reply:
👍